leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/LeshanClientBuilder.java

```diff
@@ -146,9 +152,9 @@ public LeshanClientBuilder setDecoder(LwM2mDecoder decoder) {
     }
 
     /**
-     * Set the Californium/CoAP {@link NetworkConfig}.
+     * Set the Californium/CoAP {@link Configuration}.
      */
-    public LeshanClientBuilder setCoapConfig(NetworkConfig config) {
+    public LeshanClientBuilder setCoapConfig(Configuration config) {
```

Comment:
I'm a bit concern about how to expose the `setCoapConfig(Configuration cfg)` and ` setDtlsConfig(DtlsConnectorConfig.Builder cfg)` together now :thinking: 

Because now the `DtlsConnectorConfig.Builder` pass to `setDtlsConfig` must be initialized with same `Configuration` used by `setCoapConfig`.
That means that a user can not really use only `setDtlsConfig(DtlsConnectorConfig.Builder cfg)`  


For now I have no idea to solve this.

Reply:
> Because now the DtlsConnectorConfig.Builder pass to setDtlsConfig must be initialized with same Configuration used by setCoapConfig.

That is my intention, but no must. If the Configuration is passed in the Builder, it's "cloned" and so may be modified without modifying the original Configuration. For me this is more a question of how many "really different Configurations" are good to handle.